File: collection_1/letter_count.py
#!/usr/bin/env python3
# https://dsa.cs.tsinghua.edu.cn/oj/problem.shtml?id=33
import re

if __name__ == '__main__':
    letter_count = {}
    s = input().upper()
    for c in s:
        if c == " ":
            continue
        if c in letter_count:
            letter_count[c] += 1
        else:
            letter_count[c] = 1
    letter = list(letter_count.keys())
    letter.sort()
    letter_filter = re.compile(r'^[A-Z]{1}$')
    for c in letter:
        if letter_filter.match(c):
            print("{}: {}".format(c, letter_count[c]))

File: collection_1/letter_count.cpp
#include <cstdio>
#include <cstring>
#include <cctype>

#define STR_LEN 4096
using namespace std;
// https://dsa.cs.tsinghua.edu.cn/oj/problem.shtml?id=33
int main(void)
{
    char text[STR_LEN + 1];
    gets(text);

    int cnt[26];
    for (int i = 0; i < 26; i++)
    {
        cnt[i] = 0;
    }
    int len = strlen(text);
    for (int i = 0; i < len; i++)
    {
        char c = text[i];
        if (isalpha(c))
        {
            cnt[toupper(c) - 'A']++;
        }
    }
    for (int i = 0; i < 26; i++)
    {
        if (cnt[i] != 0)
        {
            printf("%c: %d\n", i + 'A', cnt[i]);
        }
    }
    return 0;
}
